### DISCUSSION TITLE
New Component: Toasts

### DISCUSSION THREAD

**Discussion by Queen-esther01** [heroui-inc/heroui#314] [Feature requests]
Toasts are a great way of communicating with users after performing an action.
I was wondering if we could add this feature to NextUI. I believe it is a must-have 
for Libraries such as this.

Something like the image below: e.g custom toasts, toasts with icon, with image, themed toasts etc.

**Additional context**
![screenshot](https://user-images.githubusercontent.com/35095984/154034922-ac17647c-ba7d-431d-9fcd-580bedaaa9a8.png)
.

**Comment by hardikhere** [comment]
@jrgarciadev If we have designs for our toast, may I work on that?

**Comment by jrgarciadev** [comment]
Hey @hardikhere I appreciate you want to do this, however, new components are being built just for the core team since we have to make several decisions about the `API` the `Design` and so on,  would you like to work on another issue? 🙏🏻

**Comment by hardikhere** [comment]
> Hey @hardikhere I appreciate you want to do this, however, new components are being built just for the core team since we have to make several decisions about the `API` the `Design` and so on, would you like to work on another issue? 🙏🏻

Sure, I actually liked the library, can you assign me to some issue I would love to contribute.

**Comment by tzwtmll** [comment]
when will this feature be available? we really like the design of your team

**Reply by okinawaa** [reply]
I fully agree

**Comment by Moe03** [comment]
Would love this feature, very needed in any app.

**Comment by zCastleM** [comment]
Would love this feature, very needed in any app. ²

**Comment by luisarellano09** [comment]
+1

**Comment by prince272** [comment]
Would love this feature, very needed in any app. Exactly!

**Comment by redrr** [comment]
+1

**Comment by okinawaa** [comment]
Would love this feature, very needed in any app. Exactly!

**Comment by orangsed** [comment]
we need toast

**Comment by Arvin8613** [comment]
Would love this feature, very needed in my app. Exactly!

**Comment by ersanyakit** [comment]
+1

**Comment by gabamnml** [comment]
+1

**Comment by AlessandroMontenero** [comment]
+1

**Comment by thomasbrach** [comment]
+1, it would be a shame to have to use another library for toasts when NextUI is so good :)

**Comment by JuanTorchia** [comment]
I've been snooping around this thread, and honestly, I think a toast component would be awesome for NextUI. Looks like many here agree. I get that the core team has a lot on their plate with design and all that stuff, but hey, I'm here to help out. We can't let NextUI fall behind just 'cause we're missing some toasts, right? 😉

If you think I can chip in somehow, just let me know. Cheers and let's rock this!

**Comment by nirdeshnccs** [comment]
+1

**Comment by kor-bim** [comment]
+1

**Comment by Th1nhNg0** [comment]
+1

**Comment by mrafieefard** [comment]
+1

**Comment by tadinski** [comment]
toast is being missed hard

**Comment by meetmakwana7396** [comment]
+1

**Comment by truethari** [comment]
+1

**Comment by iYonga** [comment]
++++++

**Comment by ratneshmurugesan** [comment]
➕ Would like to see a toast component that is similar to the ones floating in nextui homepage

**Comment by leeonon** [comment]
+1

**Comment by ramiborni** [comment]
+1

**Comment by slysiou** [comment]
+1

**Reply by slysiou** [reply]
I can't wait. So, I am using react-toastify.
https://fkhadra.github.io/react-toastify/introduction
Works well with result of server action on nextjs 14

**Comment by cassiothacosta** [comment]
+1

**Comment by alisonjonck** [comment]
+1

**Comment by ArianSha** [comment]
+1

**Comment by fresholia** [comment]
+1

**Comment by Iqbal12333** [comment]
+1

**Comment by Martinnord** [comment]
Any updates? Toasts are pretty standard in all other UI libraries

**Comment by AlpayC** [comment]
+1 bump <3

**Comment by MadMods96** [comment]
+1

**Comment by yWilliamLima** [comment]
up

**Comment by minhthao56** [comment]
+1

**Comment by kingazcore** [comment]
+1

**Comment by e-Naeim** [comment]
+1

**Comment by rlesniak** [comment]
+2

**Comment by xiaochunyong** [comment]
+1

**Comment by SaadFarooq-Dev** [comment]
The theme in the toast follows the NextUi design philosophy in my opinion. 
https://sonner.emilkowal.ski/

**Comment by maaaathis** [comment]
+1

**Comment by niklasp** [comment]
+1

**Comment by Strangeyboy** [comment]
+1

**Comment by ekajaya740** [comment]
+1

**Comment by NerminKarapandzic** [comment]
+1

**Comment by mozhu811** [comment]
+1

**Comment by rafaellaurindo** [comment]
+1

**Comment by Jarpiino** [comment]
+69

**Comment by hifiveno1** [comment]
+10086

**Comment by DhifMlak** [comment]
any news about this feature ? +1

**Comment by harrycarp** [comment]
+1

**Comment by MikeWillis** [comment]
+1

**Comment by wingkwong** [comment]
+1

**Comment by NanaLean** [comment]
+1

**Comment by Ravikumar15081987** [comment]
+1

**Comment by gabrielferreiraa** [comment]
+1

**Comment by J4v4Scr1pt** [comment]
+1

**Comment by Xam-Mlr** [comment]
+1

**Comment by alexander-densley** [comment]
updates? +10983490812730498172093841

**Reply by slysiou** [reply]
I can't wait. So, I am using react-toastify.
https://fkhadra.github.io/react-toastify/introduction
Works well with result of server action on nextjs 14 and nextui.

**Comment by githubdontbanmeagain** [comment]
+1

**Comment by git-sujon** [comment]
Any Update on That? NextUI is incompleted with a Toast...

**Comment by satu-jovanhartono** [comment]
+1

**Comment by danwild** [comment]
+1

Shame to introduce a whole new lib to a project just for a mere slice of toast 😬

**Comment by syskin** [comment]
+1

**Comment by peter-schout** [comment]
Would love this feature, very needed in any app. Exactly! +1

**Comment by LeoLin4258** [comment]
+1

**Comment by kimhwanhoon** [comment]
Thanks guys for your work! We will be patiently waiting for the upcoming update (hopefully!)

**Reply by luohao123** [reply]
No, I don't want it, do not update

**Comment by alilishan** [comment]
+ 1

**Comment by luc-hn** [comment]
+1, I need it

**Comment by mauriciopiber** [comment]
+1

**Comment by XinBaoCode** [comment]
+1

**Comment by victormanuelrequena** [comment]
If your can't work on it because don't let to the people make it?
We need the feature

**Comment by Usercyk** [comment]
+1

**Comment by SzymonDukla** [comment]
+1000

**Comment by vitalijalbu** [comment]
needdd iittttt

**Comment by jkiss** [comment]
toast is [In Review](https://nextuioss.featurebase.app/roadmap), here is a good substitute [Notification](https://ant.design/components/notification)

**Comment by halesyy** [comment]
Didn't expect so many people wanted this feature. My workaround is going to be using toastify + NextUI cards. Toasts natively would be nice - they don't have to be complicated.

**Comment by abhishekmg** [comment]
need this feature

**Comment by gamoreli** [comment]
+1

**Comment by Gzbox** [comment]
+1

**Comment by j-vitali** [comment]
c'mon guys.. not compoleted yet this **toast**?? @jrgarciadev as I know aria spectrum already have it... it's not so difficult to implement it...

**Reply by toxsick** [reply]
Pretty demanding tone here, keep in mind you‘re getting stuff for free…

**Reply by J4v4Scr1pt** [reply]
Agree with above, you can be sure that they work as hard as they can. They also have a life you know...

**Reply by j-vitali** [reply]
ok but you cannot spend 250$ and have only some cards and inputs... as I know many people bought the **PRO version**, and this component wasn't requested today or yesterday... it has months and months...

**Reply by j-vitali** [reply]
**on Feb 15, 2022** to be precise....
today **April 10 2024**

**Reply by J4v4Scr1pt** [reply]
True, but this is surely not the only component people have asked for. They also need to prioritize other stuff. No need to take this to a negative tone, that's all we are saying. It will come, I'm using Sonner with the card component while waiting. 
Happy coding m8! 🤙

**Reply by toxsick** [reply]
@j-vitali  I guess then you don't spend 250$ and as it sounds you want the toast to be part of the free version. Also pro just came out btw...

@J4v4Scr1pt I'm also using Sonner with card and it works totally fine.

Said my piece, out of this now. LOVE

**Reply by j-vitali** [reply]
share pls your code.... thanks

**Reply by J4v4Scr1pt** [reply]
ofc, because I like thriving community, especially with this lib.
```
export function ShowToaster(toasterOptions: ToastOptions) {
	if (toasterOptions) {
		toast(
			<Card
				className={cn('w-full min-w-80  border border-solid', {
					'border-lightTurquoise bg-whiteTurquoise': toasterOptions.type === 'success',
					'border-lightRed bg-alternativeRed': toasterOptions.type === 'error',
				})}>
				<CardBody className="flex flex-row items-center justify-between p-0 px-4 py-5">
					<span
						className={cn('normal-regular', {
							'text-skobeloff': toasterOptions.type === 'success',
							'text-brick': toasterOptions.type === 'error',
						})}>
						{toasterOptions.message}
					</span>
					<Button
						isIconOnly
						radius="full"
						size="sm"
						variant="light"
						onPress={() => toast.dismiss(toasterOptions.id)}>
						<Icon iconName="cross16" size="sm" />
					</Button>
				</CardBody>
			</Card>,
			{
				id: toasterOptions.id,
				duration: 3000,
				position: toasterOptions?.position || 'top-right',
				unstyled: true,
			}
		);
	}
}
```

**Reply by j-vitali** [reply]
and how you use it? as a function or as an included componnt?

**Reply by J4v4Scr1pt** [reply]
Sorry I don't wanna sound rude.. but read the [docs ](https://sonner.emilkowal.ski/)plz.

**Reply by j-vitali** [reply]
I also don't wanna sound rude, but how the f*ck I can know that **Sonner** is a library?!

**Reply by MikeWillis** [reply]
jesus dude, it was posted on this discussion in November of last year. You gotta chill out

https://github.com/nextui-org/nextui/discussions/314#discussioncomment-7671361

**Reply by J4v4Scr1pt** [reply]
Man.. woow.. I only helped you.. I really hope you come to senses because otherwise your career in this field will be short. When you post in forums and such you are expected to behave. That's why I emphasized that I did NOT wanna come of as rude. But I'm done helping you, that's for sure. I wish you good luck with your coding and hope your anger issues goes away 🤍.

**Reply by freddyrangel** [reply]
I don't think it's out of line to expect a bit more from a company that has an open source project but makes money from a pro version. That said I think they're doing a pretty good job and I can't complain. Surely they'll get around to it

**Reply by j-vitali** [reply]
yes but, **you have time to create new PRO components** and don't have time to finish a mega requested component for 2 years??! c'mon...

**Reply by j-vitali** [reply]
and also all these PR's??! still in reviewing? the project needs a PM asap.

**Reply by freddyrangel** [reply]
Yeah NextUI lacks a lot of components we've come to expect from a UI library. But they sure look pretty don't they?

**Comment by StarkMP** [comment]
+1

**Comment by armatrix** [comment]
+1

**Comment by rromanovsky** [comment]
+1

**Comment by ponijon** [comment]
+1

**Comment by AashiqDurga** [comment]
+1

**Comment by georgep21** [comment]
+1

**Comment by zhangyinkai** [comment]
+1

**Comment by xBirahim** [comment]
+1

**Comment by Hosar** [comment]
Please add Toast component +1

**Comment by ynchen2829** [comment]
+1

**Comment by r4g309** [comment]
+1

**Comment by thallium** [comment]
+1

**Comment by idylicaro** [comment]
+1

**Comment by Inbestigator** [comment]
+1

**Comment by israelviveros** [comment]
+1

**Comment by alarv** [comment]
+1

**Comment by CMOISDEAD** [comment]
+1

**Comment by chaiyilin** [comment]
+1

**Comment by aakritip12** [comment]
+1

**Comment by LiamConan** [comment]
+1

**Comment by ravindra887** [comment]
+1

**Comment by pushduck** [comment]
+1111

**Comment by elmariachi111** [comment]
+1+1

**Comment by WatanabeThiago** [comment]
+1

**Comment by blakemulnix** [comment]
the people demand toast! +1

**Comment by bug-breeder** [comment]
+1

**Comment by lutfikrasnici1** [comment]
+1

**Comment by MiloWang2048** [comment]
+1

**Comment by hanhkim** [comment]
+++1

**Comment by dengxiwang** [comment]
+1

**Comment by japalekhin** [comment]
+3820000

**Comment by SuperSaiyr** [comment]
Because the project lacks basic component like toast and alert, I have decided to remove NextUI from my project

**Reply by leeshuzheng** [reply]
I feel like toasts would be a good addition. At the same time, NextUI's tooltips are one of the best around

**Reply by Eucotopia** [reply]
But they really lack many important components, don't they? And their development cycle is too long

**Comment by TheDanniCraft** [comment]
+1

**Comment by yasircs4** [comment]
+1

**Comment by deep-os** [comment]
+1

**Comment by Andrzej661** [comment]
+1

**Comment by kevinpiac** [comment]
+1

**Comment by shi-rudo** [comment]
+1

**Comment by PaisWillie** [comment]
+1

**Comment by FelixSelter** [comment]
+1 we need toasts

**Comment by WZhijun154** [comment]
+1

**Comment by ryosukemori** [comment]
+1

**Comment by felix2056** [comment]
+1

**Comment by vitalijalbu** [comment]
guys, just use shadcn or radix-themes. they are awesome... forgot about nextui... it's almost 1-2 years we want this damn Toast and Alert... Drawer included...

**Reply by kevinpiac** [reply]
I can understand you're pissed about it. So do I.  😆 

Using an external toast is for sure the right solution right now. 

However, I believe it's great to show our interest in this damn toast component haha.

If any Next UI maintainer comes here, this feature would be a quick win for sure.

**Reply by shi-rudo** [reply]
Even if NextUI has not implemented any toasts, it is still a very good component library. Furthermore, it is not very cumbersome to use an external toast package for it. But of course, I would also like to have a one available in NextUI :-)

**Reply by felix2056** [reply]
Not just toasts, lots of other components are missing like charts, alerts, dialogues, carousels, scrollspys among a few other things.

**Comment by yago85** [comment]
+100500

**Comment by ayoub-khemissi** [comment]
+1

**Comment by JiangYingjin** [comment]
+1

**Comment by London804** [comment]
I'm confused, this feature was requested over two years ago. If we have people that are willing to work on it why is this not being worked on?

**Comment by mxvsh** [comment]
+1

**Comment by tiagoguaro323** [comment]
such a good UI but what's goin on with the work on the project?, ppl want to help, team don't let join but also don't work on the features too :/

**Reply by London804** [reply]
There is an open PR for an Alert component. I'm told once that is done we will be able to work on this component, as there is a lot of overlap.

**Comment by QuentinDEME97** [comment]
Guys, it's 2024, i think toast and alerts are mandatory features for component library. Will it be released one day or not ?

I won't judge the fact you don't have time, etc. But can we have answers ?

**Reply by danielunited** [reply]
It's in progress. https://nextuioss.featurebase.app/roadmap

In the meantime, [react-hot-toast](https://react-hot-toast.com/) is a good alternative, and the design language is pretty much the same.

**Comment by henry-oway** [comment]
Throwing out the 400th +1 here. Need to implement a notification component!

**Comment by anuragjatofficial** [comment]
+1 ( I am so obsessed with toasts BTW )

**Comment by DevChrisMiller** [comment]
+1 for the TOAST 🍞

**Comment by MeMatty** [comment]
Another +1 here, working on my first HeroUI project after doing some stuff with MUI. This seems like a must have for a UI lib to display user feedback cleanly.

**Comment by wingkwong** [comment]
PR: https://github.com/heroui-inc/heroui/pull/4437

**Comment by wingkwong** [comment]
Closing - available in 2.7.2

**Comment by gregorioafk** [comment]
If you're experiencing style issues, it could be due to NextUI's CSS conflicting with HeroUI.
To avoid this, we recommend migrating fully to HeroUI.

Thanks for making this change, I really needed it! 🚀
